helpers/getInferredName.js

```diff
@@ -6,5 +6,7 @@ try {
 	getInferredName = Function('s', 'return { [s]() {} }[s].name;');
 } catch (e) {}
 
-var inferred = function () {};
+// eslint-disable-next-line no-new-func
+var inferred = Function('var inferred = function () {}; return inferred;')();
```

Comment:
This adds a usage of eval - a horrifically terrible bad practice that should be avoided at all costs - for a very unclear benefit.

What do you mean by "compress friendly"? Do you have any kind of reproducible metrics to compare before and after this change?

Reply:
this module is depended by storybook, when you build storybook, this module is  minified, and the last line of minified code is

```
module.exports = getInferredName && "inferred" === function() {}.name ? getInferredName : null
```

will always return null, generate bad result in same cases: https://github.com/es-shims/Symbol.prototype.description/pull/13

minifier will shorten the variable name or even remove it(in this case).

Reply:
which minifier? In that case the minifier is broken; you can't safely inline anonymous functions that are in places where name inference applies.

Reply:
webpack's builtin minifier, this is how minifier works by default.

Reply:
Has an issue been filed on webpack? Removing function names isn't a safe default.

Reply:
For the workaround here, I'd prefer to do something simpler that tricks the minifier so it doesn't try to inline `inferred` rather than use any form of eval. Perhaps `inferred;` on its own line, or something?

Reply:
I don't think this is minifier's problem, because all minifiers(uglify,babel...) works the same way

Reply:
try https://github.com/babel/minify

Reply:
https://javascript-minifier.com/
```
(function(){
  var xx=1;
  
  var infer=function (){};
  
  console.log(infer.name,xx);
  })();
```

Reply:
I understand that many minifiers might be broken; that doesn't mean they shouldn't be fixed ASAP.

Function names, and inference, are an observable part of the JS specification, and any minifier that breaks this is violating the JS spec.

Reply:
> This adds a usage of eval - a horrifically terrible bad practice that should be avoided/minimized at all costs (i realized it's used elsewhere in this file) - for a very unclear benefit.

This file uses it because it depends on computed and method properties, which are an **ES2015** feature.